src/agent/agent.js

```diff
@@ -59,9 +63,16 @@ export class Agent {
                 bannedFood: ["rotten_flesh", "spider_eye", "poisonous_potato", "pufferfish", "chicken"]
             };
 
-            if (init_message) {
-                this.handleMessage('system', init_message);
-            } else {
+            if (save_data && save_data.self_prompt) { // if we're loading memory and self-prompting was on, restart it, ignore init_message
```

Comment:
Can we handle this inside of the self prompter class? Maybe always call self_prompter.start() here but if the saved `self_prompt` variable is empty it just returns.